src/transformers/models/deberta/modeling_tf_deberta.py

```diff
@@ -0,0 +1,1518 @@
+    def __init__(self, config: DebertaConfig, **kwargs):
+        super().__init__(**kwargs)
+        self.dense = tf.keras.layers.Dense(config.pooler_hidden_size, name="deberta_pooler_dense")
+        self.dropout = StableDropout(config.pooler_dropout, name="deberta_pooler_dropout")
+        self.config = config
+
+    def call(self, hidden_states, training: bool = False):
+        # We "pool" the model by simply taking the hidden state corresponding
+        # to the first token.
+        context_token = hidden_states[:, 0]
+        context_token = self.dropout(context_token, training=training)
+        pooled_output = self.dense(context_token)
+        pooled_output = get_tf_activation(self.config.pooler_hidden_act)(pooled_output)
+        return pooled_output
+
+    @property
+    def output_dim(self) -> int:
+        return self.config.hidden_size
+
+
+class XSoftmax(tf.keras.layers.Layer):
+    """
+    Masked Softmax which is optimized for saving memory
+
+    Args:
+        input (:obj:`torch.tensor`): The input tensor that will apply softmax.
+        mask (:obj:`torch.IntTensor`): The mask matrix where 0 indicate that element will be ignored in the softmax calculation.
+        dim (int): The dimension that will apply softmax
+    """
+
+    def __init__(self, axis=-1, **kwargs):
+        super().__init__(**kwargs)
+        self.axis = axis
+
+    def call(self, inputs: tf.Tensor, mask: tf.Tensor):
+
+        rmask = tf.logical_not(tf.cast(mask, tf.bool))
+        output = tf.where(rmask, float("-inf"), inputs)
+        output = tf.nn.softmax(output, self.axis)
+        output = tf.where(rmask, 0.0, output)
+        return output
+
+
+def get_mask(input, dropout):
+    mask = tf.cast(
+        1 - tf.compat.v1.distributions.Bernoulli(probs=1 - dropout).sample(sample_shape=shape_list(input)), tf.bool
+    )
+    return mask, dropout
+
+
+@tf.custom_gradient
+def XDropout(input, local_ctx):
+    mask, dropout = get_mask(input, local_ctx)
+    scale = tf.convert_to_tensor(1.0 / (1 - dropout), dtype=tf.float32)
+    input = tf.cond(dropout > 0, lambda: tf.where(mask, 0.0, input) * scale, lambda: input)
+
+    def custom_grad(upstream_grad):
+        return tf.cond(
+            scale > 1, lambda: (tf.where(mask, 0.0, upstream_grad) * scale, None), lambda: (upstream_grad, None)
+        )
+
+    return input, custom_grad
+
+
+class StableDropout(tf.keras.layers.Layer):
+    """
+    Optimized dropout module for stabilizing the training
+
+    Args:
+        drop_prob (float): the dropout probabilities
+    """
+
+    def __init__(self, drop_prob, **kwargs):
+        super().__init__(**kwargs)
+        self.drop_prob = tf.convert_to_tensor(drop_prob, dtype=tf.float32)
+
+    def call(self, inputs: tf.Tensor, training: tf.Tensor = False):
+        if training and self.drop_prob > 0:
+            return XDropout(inputs, self.drop_prob)
+        return inputs
+
+
+class DebertaLayerNorm(tf.keras.layers.Layer):
```

Comment:
Have you tried just using the standard TF LayerNorm here? It probably doesn't make a big difference either way, but this looks very close to the built-in library module!

Reply:
I think it should be almost the same. The only difference is where to apply the eps, i.e. inside sqrt or outside of sqrt.

Reply:
changed to a built-in layer. https://github.com/huggingface/transformers/pull/12972/commits/a15bc1a588ca18d4d302882c95cb654a8b1574e2